transport/src/test/java/io/netty/channel/AdaptiveRecvByteBufAllocatorTest.java

```diff
@@ -54,6 +54,26 @@ public void rampUpBeforeReadCompleteWhenLargeDataPending() {
         allocReadExpected(handle, alloc, 8388608);
     }
 
+    @Test
+    public void memoryAllocationIntervalsTest() {
+        computingNext(512, 512);
+        computingNext(8192, 1110);
+        computingNext(8192, 1200);
+        computingNext(4096, 1300);
+        computingNext(4096, 1500);
+        computingNext(2048, 1700);
+        computingNext(2048, 1550);
+        computingNext(2048, 2000);
+        computingNext(2048, 1900);
```

Comment:
Can we also assert the guess incrementing after the read bytes increases?

Reply:
yes, But I think my unit test is correct.

Reply:
I'm not saying it's incorrect, but it only asserts in one direction.

Reply:
Maybe method ``rampUpBeforeReadCompleteWhenLargeDataPending`` contains the assertions you need.

Reply:
Point was to test going down and up to ensure the buffer is adaptive.

Reply:
@brucelwl can you add the asserts ?

Reply:
Sorry, I don't have much time to add such code. Maybe you can turn off this pr, but please fix issue https://github.com/netty/netty/issues/9549

Reply:
After playing with this a bit I feel the unit test is flawed as it does not take into consideration the index increment and decrement step.